### PR TITLE
Added cloneNode method to ParentNode object.

### PR DESCRIPTION
**Pull request by Indomian**
- Added tests for ParentNodeNodeParent.cloneNode.
 - Added tests for ParentNode.hasChildNodes (required for node cloning).

Signed-off-by: Egor Bolgov <e.bolgov@semrush.com>

### CODE REVIEW

**Comment by Indomian** [comment]
@christophehurpeau Can you merge this first so i can push branch with HTMLAnchorElement? It depends on cloneNode.

File: src/ParentNode.js [hunk 1]
```diff
@@ -312,4 +322,9 @@ export default class ParentNode extends Node {
             return value + node.textContent;
         }, '');
     }
+
+    cloneNode(deep) {
```
Comment (christophehurpeau) [inline src/ParentNode.js]: cloneNode(deep = false)

Comment (christophehurpeau) [inline src/ParentNode.js]: you can also add documentation

File: src/utils/cloneNodeHelper.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+/**
+ * Function return copy of element and if request deep copy of element
+ *
+ * @param {Node} element
+ * @param {boolean} deep
+ * @ignore
+ */
+function cloneNode(element, deep) {
```
Comment (christophehurpeau) [inline src/utils/cloneNodeHelper.js]: export function cloneNode ?

for libs/helpers/tools prefer export functions and not as default

File: src/ParentNode.js (hunk 1 above)
Comment (Indomian) [inline src/ParentNode.js]: Done.

File: src/utils/cloneNodeHelper.js (hunk 2 above)
Comment (Indomian) [inline src/utils/cloneNodeHelper.js]: Rewrite with exporting functions from library.

**Comment by christophehurpeau** [comment]
Thanks !

**Pull request merged**